app/routes/thematic/geoencoder.py

```diff
@@ -0,0 +1,216 @@
+import re
+from typing import Any, Dict, List, Optional
+
+from fastapi import APIRouter, HTTPException, Query
+from unidecode import unidecode
+
+from app.crud.versions import get_version, get_version_names
+from app.errors import RecordNotFoundError
+from app.models.pydantic.responses import Response
+from app.routes import VERSION_REGEX
+from app.routes.datasets.queries import _query_dataset_json
+
+router = APIRouter()
+
+
+@router.get(
+    "/geoencode",
+    tags=["Geoencoder"],
+    status_code=200,
+)
+async def geoencode(
+    *,
+    admin_source: str = Query(
+        "GADM", description="The source of administrative boundaries to use."
```

Comment:
Should we document that right now the only option is "GADM"?

Reply:
Done!